### ISSUE TITLE
Error Parsing Feature Flags

### ISSUE THREAD

**Issue by jedi4ever** [jszwedko/go-circleci#8]
FeatureFlags is defined as map[string]bool
But in reality they are sometimes null too

```
 "feature_flags" : {
   "trusty-beta" : true,
   "osx" : false,
   "set-github-status" : true,
   "build-prs-only" : false,
   "fleet" : null,
   "build-fork-prs" : false,
   "autocancel-builds" : false,
   "junit" : true
 },
```

**Comment by jszwedko** [comment]
Oh dear, that is unfortunate. I'm going to reach out to them first to see if that is intentional, but we may need to use a `*bool` for that type.

**Comment by jszwedko** [comment]
Actually, it looks like `null` parses as `false` fine, but they definitely have different semantics. I'll still reach out.

**Comment by jszwedko** [comment]
Created https://discuss.circleci.com/t/possible-api-bug-in-projects-for-feature-flags/8213

**Comment by jszwedko** [comment]
Got an answer in the linked discussion: it looks like `fleet` is a (nullable) `string` rather than a `bool`. I think that leaves our options as:

1. create a `struct` for `FeatureFlags` rather than using a `map[string]bool`, something like:

```
type FeatureFlags struct {
  TrustyBeta bool
  OSX bool
  ...
  fleet *string
}
```

but perhaps have a way to get a raw `map[string]interface{}` (e.g. defining `.Raw()`) to help with accessing feature flags that are added in-between `go-circleci` releases.

2. simply change the `FeatureFlags` to be `map[string]interface{}` and let the user handle the type assertions

I'll think about this a bit more, but do you have thoughts on the above @jedi4ever (or a different suggestion)?

**Comment by jszwedko** [comment]
Fixed by #20
